docs/guides/basics/parameters.md

```diff
@@ -0,0 +1,251 @@
+
+```ts
+coda.makeParameter({
+  type: coda.ParameterType.String,
+  name: "type",
+  description: "The type of cookie.",
+})
+```
+
+See [`ParamDef`][ParamDef] for the full set of properties you can define for a parameter.
+
+## Accessing parameter values
+
+At runtime, the values set by the user are passed to the formula's `execute` function as the first argument, bundled up as an array.
+
+```ts
+pack.addFormula({
+  // ...
+  parameters: [
+    coda.makeParameter({
+      type: coda.ParameterType.String,
+      name: "type",
+      description: "The type of cookie.",
+    }),
+    coda.makeParameter({
+      type: coda.ParameterType.Number,
+      name: "num",
+      description: "How many cookies.",
+    }),
+  ],
+  // ...
+  execute: async function ([type, num], context) {
+    // ...
+  },
+});
+```
+
+The order that you define the parameters determines the order they are passed into the `execute` function. The names of the parameters don't need to match the variable names you use for them in the `execute` function, but it's usually more readable to keep them the same.
+
+??? "Array destructuring"
+    In the code above, and across our other samples, we typically use [array destructuring][destructuring_assignment] to pull values out of the parameter array and assign them to variables. You could alternatively do that within the body of the `execute` function:
+
+    ```ts
+    execute: async function (parameters, context) {
+      let word = parameters[0];
+      let count = parameters[1];
+    },
+    ```
+
+
+## Optional parameters
+
+By default all parameters you define are required. To make a parameter optional simply add `optional: true` to your parameter definition. Optional parameters are shown to the user but not required in order for the formula to execute.  Optional parameters must be defined after all of the required parameters, and like required parameters their order is reflected in the Coda formula editor and the array of values passed to the `execute` function.
+
+
+```ts
+pack.addFormula({
+  // ...
+  parameters: [
+    coda.makeParameter({
+      type: coda.ParameterType.String,
+      name: "name",
+      description: "The person's name.",
+    }),
+    coda.makeParameter({
+      type: coda.ParameterType.String,
+      name: "suffix",
+      description: "A suffix, like 'MD' or 'Jr'.",
+      optional: true,
+    }),
+  ],
+  // ...
+  execute: async function ([name, suffix], context) {
+    // ...
+  },
+});
+```
+
+Optional parameters that have not been set by the user will default to the JavaScript value `undefined` in your `execute` function. When you initialize your parameter variables in the `execute` function you can assign a default value that will get used when the parameter has not been explicitly set by the user.
+
+```ts
+--8<-- "examples/formula/scream.ts"
```

Comment:
Think this example didn't render properly in the staged docs.

Reply:
Good catch, it got lost during a refactor. I've enabled strict include checking, so this doesn't happen again.